### PR TITLE
Calculating query complexity

### PR DESCRIPTION
**Pull request by BeeHiveJava**
This pull request adds a helper function to calculate query complexity independently from query execution, see #16. 

I was unsure if other tests should also use this functionality. Some don't just use the complexity value, but also errors found in ```ValidationContext``` after validating, so that would require some modification.

### CODE REVIEW

File: src/__tests__/QueryComplexity-test.ts [hunk 1]
```diff
@@ -23,6 +23,23 @@ import {
 describe('QueryComplexity analysis', () => {
   const typeInfo = new TypeInfo(schema);
 
+  it('should calculate complexity', () => {
+    const ast = parse(`
+      query {
+        variableScalar(count: -100)
```
Comment (ivome) [inline src/__tests__/QueryComplexity-test.ts]: Better use a value that is not an ignored edge case internally (negative values), so we can actually be sure the value is calculated.

Comment (BeeHiveJava) [reply src/__tests__/QueryComplexity-test.ts]: You're right, fixed.

**Comment by BeeHiveJava** [comment]
Added an example and fixed the test to take a value that actually gets calculated.
